AppIntegrity/Sources/AppIntegrity/FirebaseAppIntegrityService.swift

```diff
@@ -41,43 +42,48 @@ public final class FirebaseAppIntegrityService: AppIntegrityProvider {
     }
 
     init(vendor: AppCheckVendor,
-         providerFactory: AppCheckProviderFactory,
+         networkClient: NetworkClient,
          proofOfPossessionProvider: ProofOfPossessionProvider,
-         client: NetworkClient,
-         baseURL: URL) {
-        self.client = client
+         baseURL: URL,
+         proofTokenGenerator: ProofTokenGenerator) {
+        self.networkClient = networkClient
         self.vendor = vendor
-        self.baseURL = baseURL
         self.proofOfPossessionProvider = proofOfPossessionProvider
+        self.baseURL = baseURL
+        self.proofTokenGenerator = proofTokenGenerator
     }
-
-    public convenience init(
-        client: NetworkClient,
-        baseURL: URL,
-        proofOfPossessionProvider: ProofOfPossessionProvider
-    ) {
+    
+    public convenience init(networkClient: NetworkClient,
+                            proofOfPossessionProvider: ProofOfPossessionProvider,
+                            baseURL: URL,
+                            proofTokenGenerator: ProofTokenGenerator) {
         self.init(
             vendor: AppCheck.appCheck(),
-            providerFactory: Self.providerFactory,
+            networkClient: networkClient,
             proofOfPossessionProvider: proofOfPossessionProvider,
-            client: client,
-            baseURL: baseURL
+            baseURL: baseURL,
+            proofTokenGenerator: proofTokenGenerator
         )
     }
-
-    public func assertIntegrity() async throws -> String {
+    
+    public func assertIntegrity() async throws -> [String: String] {
         guard !isValidAttestationAvailable else {
             // nothing to do:
             throw NotImplementedError.notImplemented
         }
 
         let token = try await vendor.token(forcingRefresh: false)
-        print("vendor token: \(token.token)")
 
         do {
-            let attestation = try await fetchClientAttestation(appCheckToken: token.token)
             // TODO: DCMAW-10322 | store this locally
-            return attestation.attestationJWT
+            let attestation = try await fetchClientAttestation(appCheckToken: token.token)
```

Comment:
Wonder is there potentially better naming for the token so that we don't have `token.token`?